control/subscription_group.go

```diff
@@ -405,23 +409,53 @@ func (s *subscriptionGroup) subscribePlugins(id string,
 			"version": plg.Version(),
 			"_block":  "subscriptionGroup.subscribePlugins",
 		}).Debug("plugin subscription")
-		pool, err := s.pluginRunner.AvailablePlugins().getOrCreatePool(plg.Key())
-		if err != nil {
-			serrs = append(serrs, serror.New(err))
-			return serrs
-		}
-		pool.Subscribe(id)
-		if pool.Eligible() {
-			err = s.verifyPlugin(plg)
+		if plg.Details.Uri != nil {
+			// this is a remote plugin
+			pool, err := s.pluginRunner.AvailablePlugins().getOrCreatePool(plg.Key())
+			if pool.Count() < 1 {
+				var resp plugin.Response
+				res, err := http.Get(plg.Details.Uri.String())
+				if err != nil {
+					serrs = append(serrs, serror.New(err))
+					return serrs
+				}
+
+				body, err := ioutil.ReadAll(res.Body)
+				if err != nil {
+					serrs = append(serrs, serror.New(err))
+					return serrs
+				}
+				json.Unmarshal(body, &resp)
+				ap, err := newAvailablePlugin(resp, s.eventManager, nil)
+				if err != nil {
+					serrs = append(serrs, serror.New(err))
+					return serrs
+				}
+				pool.Insert(ap)
+			}
 			if err != nil {
 				serrs = append(serrs, serror.New(err))
 				return serrs
 			}
-			err = s.pluginRunner.runPlugin(plg.Name(), plg.Details)
+		} else {
+			pool, err := s.pluginRunner.AvailablePlugins().getOrCreatePool(plg.Key())
 			if err != nil {
 				serrs = append(serrs, serror.New(err))
 				return serrs
 			}
+			pool.Subscribe(id)
+			if pool.Eligible() && plg.Details.Uri == nil {
```

Comment:
Checking for `plg.Details.Uri == nil` isn't required here as the else case will be hit only if the Uri is nil.